<?xml version="1.0"?>

<launch>

  <arg name="drone_name"          default="ragnarok"/>
  <arg name="target_name"         default="boat_landing_platform"/>
  <arg name="rover_serial_port"   default="/dev/ttyACM1"/>
  <arg name="base_host"           default="192.168.0.143"/>
  <arg name="rover_host"          default="192.168.0.84"/>


  <!-- Load common parameters -->
  <rosparam command="load" file="$(find roscopter)/params/$(arg drone_name).yaml"/>

  <!-- PID Position Controller -->
  <node name="controller" pkg="roscopter" type="ff_cntrl_node" output="screen">
    <remap from="target_estimate" to="/boat/odom"/>
    <remap from="estimate" to="odom"/>
    <remap from="waypoint" to="high_level_command"/>
  </node>

  <!-- Estimator -->
  <rosparam command="load" file="$(find roscopter)/params/ekf.yaml" ns="estimator"/>
  <node name="estimator" type="ekf_node" pkg="roscopter" output="screen">
    <remap from="imu" to="/imu/data"/>
    <!-- <remap from="baro" to="/baro"/> -->
    <remap from="ublox_gnss" to="/PosVelEcef"/>
    <remap from="ublox_relpos" to="/RelPos"/>
  </node>

  <!-- Waypoint Manager -->
  <rosparam command="load" file="$(find roscopter)/params/waypoints.yaml"/>
  <node name="waypoint_manager" pkg="roscopter" type="waypoint_manager.py" output="screen">
    <remap from="waypoint" to="high_level_command"/>
    <remap from="state" to="odom"/>
    <remap from="base_odom" to="/boat/odom"/>
  </node>

</launch>
